source/mir/math/sum.d

```diff
@@ -1826,6 +1826,32 @@ template sum(F, Summation summation = Summation.appropriate)
         }
     }
 
+    ///
+    template sum(Range)
+        if (isIterable!Range && !isMutable!Range)
+    {
+        import core.lifetime: move;
+
+        ///
+        F sum(Range r)
+        {
+            import core.lifetime: move;
```

Comment:
Maybe we can make things simpler. Doesn't`return .sum!(F, summation)(r.lightConst);` work?

Reply:
Definitely makes it simpler. Like I said before, I don't really understand `lightConst` and it's interaction with `move`.